FestivalPlanner/films/views.py

```diff
@@ -67,9 +69,60 @@ class BaseFilmsFormView(LoginRequiredMixin, FormView):
     success_view_name = None
     post_attendance = None
     film = None
+    found_films = None
 
     def form_valid(self, form):
+        search_text = form.cleaned_data[self.SEARCH_KEY]
+        if search_text:
+            self.search_title(search_text)
+        else:
+            self.update_rating(form)
+        return super().form_valid(form)
+
+    def form_invalid(self, form):
+        self.view.unexpected_errors.extend(wrap_up_form_errors(form.errors))
+        super().form_invalid(form)
+        return self.clean_response()
+
+    def get_success_url(self):
+        errors = self.view.unexpected_errors
+        fragment = '#top' if not self.film or errors else FragmentKeeper.fragment_code(self.film.film_id)
+        return reverse(self.success_view_name) + fragment
+
+    @staticmethod
+    def clean_response():
+        return HttpResponseRedirect(reverse('films:films'))
+
+    def search_title(self, text):
+        session = self.request.session
+        initialize_log(session, action=f'Search "{text}"')
+        festival = current_festival(session)
+        found_films = []
+        start_by_film = {}
+        for film in Film.films.order_by('seq_nr').filter(festival=festival):
+            start_pos = self.film_matches_search_text(film, text)
+            if start_pos is not None:
+                found_films.append(film)
+                start_by_film[film] = start_pos
+                add_log(session, film.sort_title)
+        if found_films:
+            sorted_tuples = sorted([(f, s) for f, s in start_by_film.items()], key=itemgetter(1))
+            BaseFilmsFormView.found_films = [f for f, s in sorted_tuples]
+        else:
+            add_log(session, f'No title found starting with "{text}"')
+
+    @staticmethod
+    def film_matches_search_text(film, text):
```

Comment:
Please find a better name for `film_matches_search_text()`, which function returns a starting postion of the.given text within the sort-title of the given film.

Reply:
Done.
New name is `start_position_of_text()`.